TypeCobol/Compiler/Parser/CodeElementsLine.cs

```diff
@@ -68,34 +69,74 @@ internal void AddCodeElement(CodeElement codeElement)
         internal void ResetCodeElements()
         {
             CodeElements = null;
-            ParserDiagnostics = null;
+            _ParserDiagnostics = null;
         }
 
         /// <summary>
         /// Error and warning messages produced while parsing the source text line
         /// </summary>
-        public IList<Diagnostic> ParserDiagnostics { get; private set; }
+        public IList<Diagnostic> ParserDiagnostics
+        {
+            get { return _ParserDiagnostics; }
+        }
 
+        private IList<Diagnostic> _ParserDiagnostics;
         /// <summary>
         /// Lazy initialization of diagnostics list
         /// </summary>
         internal void AddParserDiagnostic(Diagnostic diag)
         {
             // Lazy list instantiation
-            if (ParserDiagnostics == null)
+            if (_ParserDiagnostics == null)
             {
-                ParserDiagnostics = new List<Diagnostic>();
+                _ParserDiagnostics = new List<Diagnostic>();
             }
 
-            ParserDiagnostics.Add(diag);
+            _ParserDiagnostics.Add(diag);
         }
 
         /// <summary>
-        /// Reset ParserDiagnostics property of this current line
+        /// Reset all diagnostics for the current line
         /// </summary>
         internal void ResetDiagnostics()
         {
-            ParserDiagnostics = null;
+            _ParserDiagnostics = null;
+            if (CodeElements != null)
+            {
+                foreach (var codeElement in CodeElements)
+                {
+                    codeElement.Diagnostics = null; //Delete all diganostics on every codeelement of this line
+                }
+            }
+        }
+
+        internal void UpdateDiagnositcsLine()
+        {
+            //Update ParserDiag lines index
+            if (_ParserDiagnostics != null)
+            {
+                _ParserDiagnostics = _ParserDiagnostics.Select(d =>
+                {
+                    d.Line = LineIndex+1;
```

Comment:
Why LineIndex + 1?

Reply:
Just a displaying purpose. Because ANTLR is 0 based, we need to return a +1 to diagnostics line in order to display it at the right place